lib/model/model.go

```diff
@@ -427,18 +427,13 @@ func (m *model) tearDownFolderLocked(cfg config.FolderConfiguration, err error)
 
 	m.fmut.Unlock()
 
-	// Close connections to affected devices
-	// Must happen before stopping the folder service to abort ongoing
-	// transmissions and thus allow timely service termination.
-	w := m.closeConns(cfg.DeviceIDs(), err)
-
 	for _, id := range tokens {
 		m.RemoveAndWait(id, 0)
 	}
 
 	// Wait for connections to stop to ensure that no more calls to methods
 	// expecting this folder to exist happen (e.g. .IndexUpdate).
-	w.Wait()
+	m.closeConns(cfg.DeviceIDs(), err).Wait()
```

Comment:
Will this not result in a bunch of panics and errors for non-existing folders? We have a bunch of panics in sentry for Index messages for folders that do not exist. If anything, this will make it worse.

Reply:
Preventing that is the whole point of waiting here before removing the map items protected by `fmut` below. Folders (as in the services) have nothing to do with these panics, just what the model stores about them. Changing the order of stopping folder service and closing conns is unrelated to these panics.

Reply:
So why was it done this way before?

Reply:
As there was nothing to terminate the request, waiting for a folder to stop could take a really long time (e.g. timed out connection). By closing first ourselves it was made sure that request would fail. However that's imo a hack (mine btw), when the correct thing is to abort such potentially long-time operation based on the context.